FetchRequests/Sources/Requests/PaginatingFetchDefinition.swift

```diff
@@ -41,16 +41,23 @@ public class PaginatingFetchDefinition<FetchedObject: FetchableObject>: FetchDef
 private extension InternalFetchResultsControllerProtocol {
     @MainActor
     func performPagination(
-        with paginationRequest: PaginatingFetchDefinition<FetchedObject>.PaginationRequest
+        with paginationRequest: PaginatingFetchDefinition<FetchedObject>.PaginationRequest,
+        completion: @escaping (_ hasPageResults: Bool) -> Void
     ) {
         let currentResults = self.fetchedObjects
         paginationRequest(currentResults) { [weak self] pageResults in
             guard let pageResults else {
+                completion(false)
                 return
             }
+
             performOnMainThread {
                 self?.manuallyInsert(objects: pageResults, emitChanges: true)
             }
+
+            DispatchQueue.main.async {
```

Comment:
Add comment along the lines of `// force this to run on the _next_ run loop, at which point any debounced insertions should have happened`

Reply:
And discussed offline that you could be cognizant of the debounced inserts flag and not use an async call if we're not debouncing.